### PR TITLE
fix race condition while creating buckets

### PR DESCRIPTION
**Pull request by wlan0**
By using the naming format

`br-${br.spec.bucketprefix}-${br.uid}` we get a deterministic name, but in such a way that if a br is created and deleted, and then another br is created by the same name - a new bucket is created the second time.

### CODE REVIEW

File: pkg/bucketrequest/bucketrequest.go [hunk 1]
```diff
@@ -80,27 +80,29 @@ func (b *bucketRequestListener) provisionBucketRequestOperation(ctx context.Cont
...
-	glog.Infof("Creating Bucket for BucketRequest %v", bucketRequest.Name)
+	name := bucketRequest.Spec.BucketPrefix
+	if name != "" {
+		name = name + "-"
+	}
+	name = name + string(bucketRequest.GetUID())
+	name = "br-" +  name
 
+	bucket, err := b.bucketClient.ObjectstorageV1alpha1().Buckets().Get(context.Background(), name, metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
```
Comment (rrati) [inline pkg/bucketrequest/bucketrequest.go:97]: If the bucket is found, then err == nil and we'll continue to process.  We would want to abort wouldn't we?

File: pkg/bucketrequest/bucketrequest.go [hunk 2]
```diff
@@ -80,27 +80,29 @@ func (b *bucketRequestListener) provisionBucketRequestOperation(ctx context.Cont
...
 		return util.ErrInvalidBucketClass
 	}
 
-	glog.Infof("Creating Bucket for BucketRequest %v", bucketRequest.Name)
+	name := bucketRequest.Spec.BucketPrefix
+	if name != "" {
+		name = name + "-"
+	}
+	name = name + string(bucketRequest.GetUID())
+	name = "br-" +  name
```
Comment (rrati) [inline pkg/bucketrequest/bucketrequest.go:94]: How does the prefix "br-" give us more determinism/uniqueness?  The big change seems to be using bucketRequest.GetUID vs util.GetUUID.  Am I missing something?

Comment (brahmaroutu) [reply pkg/bucketrequest/bucketrequest.go:94]: I do feel "br-" is superfluous and changes the prefix intended.

File: pkg/bucketrequest/bucketrequest.go (hunk 1 above)
Comment (brahmaroutu) [reply pkg/bucketrequest/bucketrequest.go:97]: I agree, FindBucket above returns util.ErrBucketAlreadyExists which makes the controller to discard this request.

File: pkg/bucketrequest/bucketrequest.go (hunk 2 above)
Comment (wlan0) [reply pkg/bucketrequest/bucketrequest.go:94]: sure. I can remove this

**Pull request merged**